runtime-extension-model/src/main/java/org/mule/runtime/config/internal/validation/OperationParameterDefaultValueDoesntSupportExpressions.java

```diff
@@ -0,0 +1,93 @@
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.internal.validation;
+
+import static org.mule.runtime.api.component.ComponentIdentifier.builder;
+import static org.mule.runtime.api.meta.model.parameter.ParameterGroupModel.DEFAULT_GROUP_NAME;
+import static org.mule.runtime.ast.api.util.ComponentAstPredicatesFactory.currentElemement;
+import static org.mule.runtime.ast.api.validation.Validation.Level.ERROR;
+import static org.mule.runtime.ast.api.validation.ValidationResultItem.create;
+
+import static java.lang.String.format;
+
+import org.mule.runtime.api.component.ComponentIdentifier;
+import org.mule.runtime.ast.api.ArtifactAst;
+import org.mule.runtime.ast.api.ComponentAst;
+import org.mule.runtime.ast.api.ComponentParameterAst;
+import org.mule.runtime.ast.api.validation.Validation;
+import org.mule.runtime.ast.api.validation.ValidationResultItem;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+/**
+ * Expressions are forbidden for optional parameter default values.
+ *
+ * @since 4.5
+ */
+public class OperationParameterDefaultValueDoesntSupportExpressions implements Validation {
+
+  private static final String DEFAULT_EXPRESSION_PREFIX = "#[";
+  private static final String DEFAULT_EXPRESSION_SUFFIX = "]";
+
+  private static final String OPERATION = "operation";
+  private static final String OPTIONAL_PARAMETER = "optional-parameter";
+  private static final ComponentIdentifier OPTIONAL_PARAMETER_IDENTIFIER =
+      builder().namespace(OPERATION).name(OPTIONAL_PARAMETER).build();
+
+  @Override
+  public String getName() {
+    return "No expressions in optional parameter default value";
+  }
+
+  @Override
+  public String getDescription() {
+    return "Expressions are forbidden for optional parameter default values";
+  }
+
+  @Override
+  public Level getLevel() {
+    return ERROR;
+  }
+
+  @Override
+  public Predicate<List<ComponentAst>> applicable() {
+    return currentElemement(component -> component.getIdentifier().equals(OPTIONAL_PARAMETER_IDENTIFIER));
+  }
+
+  @Override
+  public Optional<ValidationResultItem> validate(ComponentAst component, ArtifactAst artifact) {
+    ComponentParameterAst defaultValueAst = component.getParameter(DEFAULT_GROUP_NAME, "defaultValue");
+    Optional<String> defaultValue = defaultValueAst.getValue().getValue();
+    if (isExpression(defaultValue)) {
+      return validationFailed(component);
+    } else {
+      return validationOk();
+    }
+  }
+
+  private Optional<ValidationResultItem> validationFailed(ComponentAst component) {
+    return component.getParameter(DEFAULT_GROUP_NAME, "name").getValue().getValue()
+        .map(name -> create(component, this,
+                            format("An expression was given for 'defaultValue' of the optional parameter '%s'", name)));
```

Comment:
The fact that this validator will return Optional.empty() if there is no "name" on the "optional-parameter" worries me a little bit.
Just a little bit, because I know the name is mandatory from the XSD validations already, but...
Do you think we should handle it just in case?

Reply:
Yeah, either that or create a `ValidationResultItem` specifying the issue without stating the name...
I don't know what happens if an `IllegalStateException` is thrown from those methods.
Normally I would say whatever is best for the developer to identify the issue, but in this case, if this happens it means there is an error in our end when wiring up the validations (because a non-existent name should trigger earlier). So it would be us who need to catch it on a test and understand what happened.

Reply:
Added the IllegalStateException